### PR TITLE
Ability to retrieve address in both formats cash and legacy via API (closes #1590)

### LINKED ISSUE #1590: Ability to retrieve address in both formats cash and legacy via API

### PR DESCRIPTION
**Pull request by dinesh-skyach**
Ability to retrieve address in both formats cash and legacy via API #1590

### CODE REVIEW

**Comment by ysv** [comment]
@dinesh-skyach what if currency doesn’t support cash address?

**Comment by dinesh-skyach** [comment]
@ysv Yes, we need this validation too. Thanks

File: app/api/api_v2/validations.rb [hunk 1]
```diff
@@ -40,5 +40,16 @@ def validate_param!(name, params)
              message: 'should be less than to.'
       end
     end
+
+    class CurrencyAddressFormat < Grape::Validations::Base
```
Comment (ysv) [inline app/api/api_v2/validations.rb]: @dinesh-skyach Can you name this class `ValidateCurrencyAddressFormat`

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/api_v2/validations.rb (hunk 1 above)
Comment (dinesh-skyach) [reply app/api/api_v2/validations.rb]: @ysv done.

File: spec/api/api_v2/deposits_spec.rb [hunk 2]
```diff
@@ -84,22 +84,47 @@
...
+  describe 'GET /api/v2/deposit_address' do
 
-    describe 'GET /api/v2/deposit_address' do
+    context 'failed' do
       it 'validates currency' do
         api_get '/api/v2/deposit_address', params: { currency: :usd }, token: token
         expect(response).to have_http_status 422
         expect(response.body).to eq '{"error":{"code":1001,"message":"currency does not have a valid value"}}'
       end
+
```
Comment (ysv) [inline spec/api/api_v2/deposits_spec.rb:97]: I think will brake validation
```ruby
api_get '/api/v2/deposit_address', params: { currency: :btc, address_format: 'legacy' }, token: token
```

**Review by ysv — CHANGES_REQUESTED** [review]

File: spec/api/api_v2/deposits_spec.rb (hunk 2 above)
Comment (dinesh-skyach) [reply spec/api/api_v2/deposits_spec.rb:97]: @ysv All specs have passed. So, I don't think this will break validation.

Comment (ysv) [reply spec/api/api_v2/deposits_spec.rb:97]: @dinesh-skyach It's specs which you are missing
Double check please

**Review by ysv — APPROVED** [review]

**Pull request merged**